src/control/lib/control/printers.go

```diff
@@ -192,7 +193,85 @@ func printNvmeControllers(controllers storage.NvmeControllers, out io.Writer, op
 	return nil
 }
 
-func printScmModules(modules storage.ScmModules, out io.Writer, opts ...PrintConfigOption) error {
+func PrintNvmeControllerSummary(nvme *storage.NvmeController, out io.Writer, opts ...PrintConfigOption) error {
+	_, err := fmt.Fprintf(out, "PCI:%s Model:%s FW:%s Socket:%d Capacity:%s\n",
+		nvme.PciAddr, nvme.Model, nvme.FwRev, nvme.SocketID, humanize.Bytes(nvme.Capacity()))
+	return err
+}
+
+func PrintNvmeControllerHealth(stat *storage.NvmeControllerHealth, out io.Writer, opts ...PrintConfigOption) error {
+	if stat == nil {
+		fmt.Fprintln(out, "Health Stats Unavailable")
+		return nil
+	}
+
+	fmt.Fprintln(out, "Health Stats:")
+
+	iw := txtfmt.NewIndentWriter(out)
```

Comment:
there seem to be other exported functions in that file that write to output and don't wrap a txtfmt.ErrWriter, what's the criteria for needing one?

Reply:
Any function that takes an io.Writer which hasn't already been wrapped with an ErrWriter should create one. This will ensure that any errors encountered while writing aren't ignored. Writing to stdout is unlikely to fail of course, but the point of accepting an io.Writer is that it could be anything (buffer, file, etc).

If there are existing printers that aren't using ErrWriter it's most likely due to oversight or because a previously-private printer got promoted to public without making the adjustment.